engine/table/src/test/java/io/deephaven/engine/table/impl/QueryTableFloatingPointZeroTest.java

```diff
@@ -0,0 +1,599 @@
+        final Table t2 = newTable(floatCol("KeyPos", 0.0f), floatCol("Pos", 0.0f));
+        for (Table out : Arrays.asList(
+                t1.naturalJoin(t2, "KeyNeg=KeyPos"),
+                t2.naturalJoin(t1, "KeyPos=KeyNeg"),
+                t1.exactJoin(t2, "KeyNeg=KeyPos"),
+                t2.exactJoin(t1, "KeyPos=KeyNeg"),
+                t1.join(t2, "KeyNeg=KeyPos"),
+                t2.join(t1, "KeyPos=KeyNeg"),
+                t1.aj(t2, "KeyNeg>=KeyPos"),
+                t2.aj(t1, "KeyPos>=KeyNeg"))) {
+            final long key = oneKey(out);
+            floatToBitsEquals(out, "KeyNeg", key, -0.0f);
+            floatToBitsEquals(out, "KeyPos", key, 0.0f);
+            floatToBitsEquals(out, "Neg", key, -0.0f);
+            floatToBitsEquals(out, "Pos", key, 0.0f);
+        }
+    }
+
+    @Test
+    public void testDoubleJoinKey() {
+        final Table t1 = newTable(doubleCol("KeyNeg", -0.0), doubleCol("Neg", -0.0));
+        final Table t2 = newTable(doubleCol("KeyPos", 0.0), doubleCol("Pos", 0.0));
+        for (Table out : Arrays.asList(
+                t1.naturalJoin(t2, "KeyNeg=KeyPos"),
+                t2.naturalJoin(t1, "KeyPos=KeyNeg"),
+                t1.exactJoin(t2, "KeyNeg=KeyPos"),
+                t2.exactJoin(t1, "KeyPos=KeyNeg"),
+                t1.join(t2, "KeyNeg=KeyPos"),
+                t2.join(t1, "KeyPos=KeyNeg"),
+                t1.aj(t2, "KeyNeg>=KeyPos"),
+                t2.aj(t1, "KeyPos>=KeyNeg"))) {
+            final long key = oneKey(out);
+            doubleToBitsEquals(out, "KeyNeg", key, -0.0);
+            doubleToBitsEquals(out, "KeyPos", key, 0.0);
+            doubleToBitsEquals(out, "Neg", key, -0.0);
+            doubleToBitsEquals(out, "Pos", key, 0.0);
+        }
+    }
+
+    @Test
+    public void testFloatJoinMultiKey() {
+        final Table t1 = newTable(intCol("Key1", 0), floatCol("KeyNeg", -0.0f), floatCol("Neg", -0.0f));
+        final Table t2 = newTable(intCol("Key1", 0), floatCol("KeyPos", 0.0f), floatCol("Pos", 0.0f));
+        for (Table out : Arrays.asList(
+                t1.naturalJoin(t2, "Key1,KeyNeg=KeyPos"),
+                t2.naturalJoin(t1, "Key1,KeyPos=KeyNeg"),
+                t1.exactJoin(t2, "Key1,KeyNeg=KeyPos"),
+                t2.exactJoin(t1, "Key1,KeyPos=KeyNeg"),
+                t1.join(t2, "Key1,KeyNeg=KeyPos"),
+                t2.join(t1, "Key1,KeyPos=KeyNeg"),
+                t1.aj(t2, "Key1,KeyNeg>=KeyPos"),
+                t2.aj(t1, "Key1,KeyPos>=KeyNeg"),
+                t1.naturalJoin(t2, "KeyNeg=KeyPos,Key1"),
+                t2.naturalJoin(t1, "KeyPos=KeyNeg,Key1"),
+                t1.exactJoin(t2, "KeyNeg=KeyPos,Key1"),
+                t2.exactJoin(t1, "KeyPos=KeyNeg,Key1"),
+                t1.join(t2, "KeyNeg=KeyPos,Key1"),
+                t2.join(t1, "KeyPos=KeyNeg,Key1"),
+                t1.aj(t2, "KeyNeg=KeyPos,Key1"),
+                t2.aj(t1, "KeyPos=KeyNeg,Key1"))) {
```

Comment:
You actually need/want one more key column for AJ in each of your tests. There's only hashing on the first n-1 key columns, so the single-key tests are testing 0 hashed keys, and the multi-key tests are testing 1.